src/offset_date_time.rs

```diff
@@ -287,6 +287,29 @@ impl OffsetDateTime {
         )
     }
 
+    /// Get the [`PrimitiveDateTime`] in the stored offset.
+    pub(crate) const fn date_time(self) -> PrimitiveDateTime {
+        let mut second = self.utc_datetime.second() as i8 + self.offset.seconds_past_minute();
+        let mut minute = self.utc_datetime.minute() as i8 + self.offset.minutes_past_hour();
+        let mut hour = self.utc_datetime.hour() as i8 + self.offset.whole_hours();
+        let (mut year, mut ordinal) = self.utc_datetime.date.to_ordinal_date();
+
+        cascade!(second in 0..60 => minute);
+        cascade!(minute in 0..60 => hour);
+        cascade!(hour in 0..24 => ordinal);
+        cascade!(ordinal => year);
+
+        let date = Date::__from_ordinal_date_unchecked(year, ordinal);
+        let time = Time::__from_hms_nanos_unchecked(
+            hour as _,
+            minute as _,
+            second as _,
+            self.nanosecond(),
+        );
+
+        PrimitiveDateTime::new(date, time)
+    }
+
```

Comment:
There's `PrimitiveDateTime::utc_to_offset` and `PrimitiveDateTime::offset_to_utc`, which are internal methods that should be self-explanatory.